ext/dom/node.c

```diff
@@ -1991,12 +2484,31 @@ PHP_METHOD(DOMNode, getLineNo)
 /* {{{ URL: https://dom.spec.whatwg.org/#dom-node-contains
 Since:
 */
-PHP_METHOD(DOMNode, contains)
+static void dom_node_contains(INTERNAL_FUNCTION_PARAMETERS, zval *other)
 {
-	zval *other, *id;
+	zval *id;
 	xmlNodePtr otherp, thisp;
 	dom_object *unused_intern;
 
+	ZEND_ASSERT(other != NULL);
+
+	DOM_GET_OBJ(otherp, other, xmlNodePtr, unused_intern);
+	DOM_GET_THIS_OBJ(thisp, id, xmlNodePtr, unused_intern);
+
+	do {
+		if (otherp == thisp) {
+			RETURN_TRUE;
+		}
+		otherp = otherp->parent;
+	} while (otherp);
+
+	RETURN_FALSE;
+}
+
```

Comment:
Wouldn't it make more sense to pass the necessary context and return a bool and have the call site be `RETURN_BOOL()`?